.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,29 @@
+name: Rust
+
+on:
+  [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install circom
+      run: |
+        git clone https://github.com/iden3/circom.git
+        cd circom
+        cargo build --release
+        cargo install --path circom
+    - name: Build
+      run: ./scripts/build_ios.sh simulator debug
+    - name: Run core tests
+      run: cd mopro-core/ && cargo test --verbose
+    - name: Run ffi tests
+      run: cd mopro-ffi/ && cargo test --verbose
```

Comment:
This might be a bit slow with current e2e tests, but we can iterate on this later. Don't need CI to block merging either right now